src/garage/sampler/rl2_sampler.py

```diff
@@ -0,0 +1,195 @@
+        self._vec_env = None
+        self._env_spec = self.env.spec
+
+    def start_worker(self):
+        """Start workers."""
+        n_envs = self._n_envs
+        assert n_envs >= self._meta_batch_size, (
+            'Number of vectorized environments '
+            'should be at least meta_batch_size.')
+        assert n_envs % self._meta_batch_size == 0, (
+            'Number of vectorized '
+            'environments should be a multiple of meta_batch_size.')
+        envs_per_worker = n_envs // self._meta_batch_size
+
+        tasks = self.env.sample_tasks(self._meta_batch_size)
+        vec_envs = []
+        for task in tasks:
+            vec_env = pickle.loads(pickle.dumps(self.env))
+            vec_env.set_task(task)
+            vec_envs.extend([vec_env] * envs_per_worker)
+        seed0 = deterministic.get_seed()
+        if seed0 is not None:
+            for (i, e) in enumerate(vec_envs):
+                e.seed(seed0 + i)
+
+        self._vec_env = VecEnvExecutor(
+            envs=vec_envs, max_path_length=self.algo.max_path_length)
+
+    def shutdown_worker(self):
+        """Shutdown workers."""
+        self._vec_env.close()
+
+    # pylint: disable=too-many-statements
+    def obtain_samples(self, itr, batch_size=None, whole_paths=True):
+        """Sample the policy for new trajectories.
+
+        Args:
+            itr (int): Iteration number.
+            batch_size (int): Number of samples to be collected. If None,
+                it will be default [algo.max_path_length * n_envs].
+            whole_paths (bool): Whether return all the paths or not. True
+                by default. It's possible for the paths to have total actual
+                sample size larger than batch_size, and will be truncated if
+                this flag is true.
+
+        Returns:
+            list[dict]: Sample paths.
+
+        Note:
+            Each path is a dictionary, with keys and values as following:
+                * observations: numpy.ndarray with shape :math:`[N, S^*]`
+                * actions: numpy.ndarray with shape :math:`[N, S^*]`
+                * rewards: numpy.ndarray with shape :math:`[N, S^*]`
+                * env_infos: A dictionary with each key representing one
+                  environment info, value being a numpy.ndarray with shape
+                  :math:`[N, S^*]`. One example is "ale.lives" for atari
+                  environments.
+                * agent_infos: A dictionary with each key representing one
+                  agent info, value being a numpy.ndarray with shape
+                  :math:`[N, S^*]`. One example is "prev_action", which is used
+                  for recurrent policy as previous action input, merged with
+                  the observation input as the state input.
+
+        """
+        logger.log('Obtaining samples for iteration %d...' % itr)
+
+        if batch_size is None:
+            batch_size = (self._episode_per_task * self.algo.max_path_length *
```

Comment:
Shall we force `batch_size` to be equal to `self._episode_per_task * self.algo.max_path_length * self._n_envs` in this context?

Reply:
Yes, that's a good idea. How we should enforce it is another issue tho, since batch_size is a required argument. If we make it optional / none, we can set it to `self._episode_per_task * self.algo.max_path_length * self._n_envs`

Reply:
What about just `assert` it